src/utils/utils.rs

```diff
@@ -151,23 +151,35 @@ pub(crate) fn download_file_with_resume(
     match download_file_(url, path, hasher, resume_from_partial, notify_handler) {
         Ok(_) => Ok(()),
         Err(e) => {
+            let mut maybe_io_error = None;
```

Comment:
Do we need the temp variable at all? This looks like it would be amenable to an early exit approach instead.

Reply:
This makes sense, we should just early return in the match arm.